routing.rst

```diff
@@ -1364,38 +1366,100 @@ Route alias allow you to have multiple name for the same route:
 
         return static function (RoutingConfigurator $routes): void {
-            $routes->alias('new_route_name', 'original_route_name');
+            $routes->add('product_show', '/product/{id}')
+                    ->controller('App\Controller\ProductController::show');
         };
 
-In this example, both ``original_route_name`` and ``new_route_name`` routes can
+Now, let's say you want to create a new route called ``product_details``
+that acts exactly the same as ``product_show``.
+
+Instead of duplicating the original route, you can create an alias for it. You can do this as follows:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/routes.yaml
+        product_show:
+            path: /some-path
+            controller: App\Controller\ProductController::show
+
+        product_details:
+            # "alias" option refers to the name of the route declared above
+            alias: product_show
+
+    .. code-block:: xml
+
+        <!-- config/routes.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <routes xmlns="http://symfony.com/schema/routing"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/routing
+                https://symfony.com/schema/routing/routing-1.0.xsd">
+
+            <route id="product_show" path="/product/{id}" controller="App\Controller\ProductController::show"/>
+            <!-- "alias" attribute value refers to the name of the route declared above -->
+            <route id="product_details" alias="product_show"/>
+        </routes>
+
+    .. code-block:: php
+
+        // config/routes.php
+        use Symfony\Component\Routing\Loader\Configurator\RoutingConfigurator;
+
+        return static function (RoutingConfigurator $routes): void {
+            $routes->add('product_show', '/product/{id}')
+                    ->controller('App\Controller\ProductController::show');
+            // second argument refers to the name of the route declared above
+            $routes->alias('product_details', 'product_show');
+        };
+
+In this example, both ``product_show`` and ``product_details`` routes can
 be used in the application and will produce the same result.
 
 .. _routing-alias-deprecation:
 
 Deprecating Route Aliases
 ~~~~~~~~~~~~~~~~~~~~~~~~~
 
-If some route alias should no longer be used (because it is outdated or
-you decided not to maintain it anymore), you can deprecate its definition:
+Route aliases can be used to provide backward compatibility for routes that
+have been renamed.
+
+Now, let's say you want to replace the ``product_show`` route in favor of
+``product_details`` and mark the old one as deprecated.
+
+In the previous example, the alias ``product_details`` was pointing to
+``product_show`` route.
+
+As you want to deprecate the ``product_show`` route, so let's invert the alias as follows
+to be able to mark it as deprecated using the ``deprecated`` option:
```

Comment:
Besides the minor comments, this is the only part which has a lot of complexity and back in forth imho, can we shorten this somehow and more "straightforward"?
Thanks

Reply:
I understand your point.
Here is a suggestion:

To mark the ``product_show`` route as deprecated, you need to "switch" the alias.
The ``product_show`` become the alias, and will now point to the ``product_details`` route.
This way, the ``product_show`` alias could be deprecated.

Reply:
Much better and shorter. Love it ❤️